grpcx/server_test.go

```diff
@@ -104,6 +104,29 @@ func TestWithRequestResponseLogs(t *testing.T) {
 	})
 }
 
+func TestWithStructuredPanicLogs(t *testing.T) {
+	t.Parallel()
+	a := assert.New(t)
+
+	t.Run("panic handler works as expected", func (t *testing.T) {
+		w := bytes.NewBufferString("")
+		logger := logx.New(logx.WriterOpt(w))
+		blankExitFunc := func() {}
+		interceptor := grpcx.WithStructuredPanicLogs(grpcx.NewRecoveryHandler(logger, grpcx.SetExitFunction(blankExitFunc)))
+
+		ctx := context.Background()
+		info := &grpc.UnaryServerInfo{FullMethod: method}
+		panickingHandler := grpc.UnaryHandler(func(ctx context.Context, req interface{}) (interface{}, error) {
+			panic("Here's Johnny!")
+		})
+
+		_, err := interceptor(ctx, exampleRequest, info, panickingHandler)
+
+		a.NoError(err)
+		a.Contains(w.String(), "Here's Johnny!")
+	})
+}
```

Comment:
i'd focus the test in RecoveryHandler, WithStructuredPanicLogs does barely nothing